### PR TITLE
Add Black formatting and linting

### PR DESCRIPTION
**Pull request by ajmaddaford**
### What is the context of this PR?
Adds Black code formatting and linting. The build will fail unless the code has been formatted with Black. 

The Pipfile fixes on a version of black as it's a pre-release and
we don't want to enable prereleases for all modules. See https://github.com/LibraryOfCongress/inside-baseball/commit/54bfb1d5c3f4844854a528992ea47967c1d9b881
and https://github.com/pypa/pipenv/issues/1760.

### How to review 
- Review the code formatting
- Test the formatting with `yarn format`
- Test the linting with `./scripts/run_lint.sh`

### CODE REVIEW

File: scripts/run_lint.sh [hunk 1]
```diff
@@ -28,6 +28,9 @@ pylint --reports=n --output-format=colorized --rcfile=.pylintrc -j 0 ./app ./tes
 # http://stackoverflow.com/questions/6626351/how-to-extract-bits-from-return-code-number-in-bash
 display_result $? 2 "Pylint linting check"
 
+black -S --check app
+display_result $? 1 "Python code formatting check"
+
```
Comment (bitdivision) [inline scripts/run_lint.sh:33]: I wonder if it's worth adding a pre-commit hook for this?

I do that on some other projects and it works quite well. I know some people really don't like it though.

Just needs a `pre-commit-config.yaml` with:
```
repos:
-   repo: https://github.com/ambv/black
    rev: stable
    hooks:
    - id: black
      language_version: python3
```

**Review by bitdivision — APPROVED** [review]
Have been wanting to do this for ages. Nice!

**Comment by MebinAbraham** [comment]
Flake 8 not happy. `C812 missing trailing comma.`

**Review by MebinAbraham — COMMENTED** [review]
Do we not also want to apply this to python files found in the `tests` folder to ensure those are consistent also?

**Comment by ajmaddaford** [comment]
> Flake 8 not happy. `C812 missing trailing comma.`

For anyone else looking at this it was because `flake8-commas` needs to be uninstalled, you can run `pipenv clean` to remove packages not in the Pipfile.

**Review by JamesGardiner — APPROVED** [review]

**Comment by ajmaddaford** [comment]
Waiting on https://github.com/ONSdigital/eq-survey-runner/pull/2045 to merge first as otherwise that pull request will be merge hell 🔥

**Review by jonnyshaw89 — APPROVED** [review]
Looks Awesome! 🎉

**Pull request merged**